src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -487,11 +488,16 @@ static MvcCoreLoggerExtensions()
                19,
                "Could not bind to model with name '{ModelName}' and type '{ModelType}' as the request did not have a content type of either 'application/x-www-form-urlencoded' or 'multipart/form-data'.");
 
-            _cannotCreateHeaderModelBinder = LoggerMessage.Define<Type>(
+            _cannotCreateHeaderModelBinderCompatVersion_2_0 = LoggerMessage.Define<Type>(
                LogLevel.Debug,
                20,
```

Comment:
This id should be higher than any other value in this file.

Reply:
I am using the same Id because this log message is a temporary one and would be removed in 3.0 version. Also it is mutually exclusive with the other one in this PR. So we should be fine?

Reply:
Not sure what the issue is here - log event ids only need to be unique within a category name. It's always confusing no matter what, though.